var/spack/repos/builtin/packages/dbcsr/package.py

```diff
@@ -58,28 +73,27 @@ def cmake_args(self):
         if len(spec.variants['cuda_arch'].value) > 1:
             raise InstallError("dbcsr supports only one cuda_arch at a time")
 
-        if ('+openmp' in self.spec
-            and '^openblas' in self.spec
-            and '^openblas threads=openmp' not in self.spec):
-            raise InstallError(
-                '^openblas threads=openmp required for dbcsr+openmp')
+        if len(spec.variants['amdgpu_target'].value) > 1:
+            raise InstallError("dbcsr supports only one amdgpu_arch at a time")
 
         args = [
+            '-DCMAKE_VERBOSE_MAKEFILE=ON',
             '-DUSE_SMM=%s' % ('libxsmm' if 'smm=libxsmm' in spec else 'blas'),
-            '-DUSE_MPI=%s' % ('ON' if '+mpi' in spec else 'OFF'),
-            '-DUSE_OPENMP=%s' % (
-                'ON' if '+openmp' in spec else 'OFF'),
+            self.define_from_variant('USE_MPI', 'mpi'),
+            self.define_from_variant('USE_OPENMP', 'openmp'),
             # C API needs MPI
-            '-DWITH_C_API=%s' % ('ON' if '+mpi' in spec else 'OFF'),
+            self.define_from_variant('WITH_C_API', 'mpi'),
             '-DBLAS_FOUND=true',
             '-DBLAS_LIBRARIES=%s' % (spec['blas'].libs.joined(';')),
             '-DLAPACK_FOUND=true',
             '-DLAPACK_LIBRARIES=%s' % (spec['lapack'].libs.joined(';')),
             '-DWITH_EXAMPLES=ON',
-            '-DBUILD_SHARED_LIBS=%s' % ('ON' if '+shared' in spec else 'OFF'),
+            self.define_from_variant('BUILD_SHARED_LIBS', 'shared'),
+            self.define_from_variant('USE_HIP', 'rocm'),
+            self.define_from_variant('USE_CUDA', 'cuda'),
         ]
 
-        if '+cuda' in self.spec:
+        if self.spec.satisfies('+cuda'):
```

Comment:
is that now best practise (I don't know)?

Reply:
Yeah, from a slack converstation, `x in spec` is not restricted to the root spec. So if a dependency enables this flag, it returns true too, whereas `.satisfies(...)` considers the root spec only. (I haven't checked the source if this is really the case).

Reply:
hmmok, yet another thing Spack should mention a bit more prominent in their dev docs (if not added since checked last), since there are more matches for `in self.spec` than for `spec.satisfies` when grepping the packages, including things like `@x.y.z` which could then potentially match the version of other packages as well